presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -0,0 +1,532 @@
+
+import com.esri.core.geometry.Envelope;
+import com.esri.core.geometry.MultiPath;
+import com.esri.core.geometry.MultiPoint;
+import com.esri.core.geometry.OperatorContains;
+import com.esri.core.geometry.OperatorCrosses;
+import com.esri.core.geometry.OperatorDisjoint;
+import com.esri.core.geometry.OperatorEquals;
+import com.esri.core.geometry.OperatorIntersects;
+import com.esri.core.geometry.OperatorOverlaps;
+import com.esri.core.geometry.OperatorSimpleRelation;
+import com.esri.core.geometry.OperatorTouches;
+import com.esri.core.geometry.OperatorWithin;
+import com.esri.core.geometry.Point;
+import com.esri.core.geometry.Polygon;
+import com.esri.core.geometry.SpatialReference;
+import com.esri.core.geometry.ogc.OGCGeometry;
+import com.esri.core.geometry.ogc.OGCLineString;
+import com.esri.core.geometry.ogc.OGCMultiLineString;
+import com.esri.core.geometry.ogc.OGCPoint;
+import com.esri.core.geometry.ogc.OGCPolygon;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlNullable;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.type.StandardTypes;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import java.lang.reflect.Method;
+
+import static com.facebook.presto.geospatial.GeometryUtils.deserialize;
+import static com.facebook.presto.geospatial.GeometryUtils.serialize;
+
+public final class GeoFunctions
+{
+    private static final String LINE = "LineString";
+    private static final String MULTILINE = "MultiLineString";
+    private static final String MULTIPOINT = "MultiPoint";
+    private static final String POLYGON = "Polygon";
+    private static final String POINT = "Point";
+    private static final String GEOMETRY = "Geometry";
+
+    private GeoFunctions() {}
+
+    @Description("Construct Line Geometry")
+    @ScalarFunction
+    @SqlType(GEOMETRY)
+    public static Slice stLine(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        OGCGeometry line = OGCGeometry.fromText(input.toStringUtf8());
+        line.setSpatialReference(null);
+        return serialize(line);
+    }
+
+    @Description("Construct Point Geometry")
+    @ScalarFunction
+    @SqlType(GEOMETRY)
+    public static Slice stPoint(@SqlType(StandardTypes.DOUBLE) double longitude, @SqlType(StandardTypes.DOUBLE) double latitude)
+    {
+        OGCGeometry geometry = OGCGeometry.createFromEsriGeometry(new Point(longitude, latitude), null);
+        return serialize(geometry);
+    }
+
+    @Description("Construct Polygon Geometry")
+    @ScalarFunction
+    @SqlType(GEOMETRY)
+    public static Slice stPolygon(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        OGCGeometry polygon = OGCGeometry.fromText(input.toStringUtf8());
+        polygon.setSpatialReference(null);
+        return serialize(polygon);
+    }
+
+    @Description("Returns area of input polygon")
```

Comment:
The implementation works for both polygons and multi-polygons. Perhaps, copy the documentation from PostGIS:

http://postgis.org/docs/ST_Area.html

Returns the area of the geometry if it is a polygon or multi-polygon.